scipy/special/c_misc/misc.h

```diff
@@ -1,6 +1,10 @@
 #ifndef C_MISC_MISC_H
 #define C_MISC_MISC_H
 
+#if __GNUC__ >= 4
+#pragma GCC visibility push(hidden)
+#endif
+
```

Comment:
Just to clarify this block as an example to see if I understand -- the compiler directive is used instead of `static` declarations because other modules in scipy need to access these functions, but we can still prevent polluting the "dynamic" namespace (i.e., that of a user importing scipy) with these objects by doing this?